src/utils/rarible.js

```diff
@@ -0,0 +1,178 @@
+import deepEqual from 'deep-equal';
+import { TypedDataUtils } from 'eth-sig-util';
+import { bufferToHex } from 'ethereumjs-util';
+import Web3 from 'web3';
+import { supportedChains } from './chain';
+
+import { ipfsPrePost, ipfsJsonPin, getNftMeta } from './metadata';
+import { raribleHashMaker } from './proposalUtils';
+
+export const buildEncodeOrder = args => {
+  const salt = Math.floor(Math.random() * 1000);
+  return {
+    type: 'RARIBLE_V2',
+    maker: args.minionAddress,
+    make: {
+      assetType: {
+        assetClass: 'ERC721',
+        contract: args.nftContract,
+        tokenId: args.tokenId,
+      },
+      value: '1',
+    },
+    take: {
+      assetType: {
+        assetClass: 'ERC20',
+        contract: args.tokenAddress,
+      },
+      value: args.price,
+    },
+    data: {
+      dataType: 'RARIBLE_V2_DATA_V1',
+      payouts: [],
+      originFees: [],
+    },
+    salt,
+    start: args.startDate,
+    end: args.endDate,
+  };
+};
+
+export const encodeOrder = async (order, daochain) => {
+  const url = `${supportedChains[daochain].rarible.api_url}/order/encoder/order`;
+  try {
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(order),
+    });
+    return response.json();
+  } catch (err) {
+    throw new Error(err);
+  }
+};
+
+export const createOrder = async (order, daochain) => {
+  const url = `${supportedChains[daochain].rarible.api_url}/order/orders`;
+  try {
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(order),
+    });
+    return response.json();
+  } catch (err) {
+    throw new Error(err);
+  }
+};
+
+export const getOrderByItem = async (contract, tokenId, maker, daochain) => {
+  const url = `${supportedChains[daochain].rarible.api_url}/order/orders/sell/byItem`;
+  const params = `?contract=${contract}&tokenId=${tokenId}&maker=${maker}`;
+  try {
+    const response = await fetch(`${url}${params}`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+    return response.json();
+  } catch (err) {
+    throw new Error(err);
```

Comment:
are these being caught in the component?

we don't want the app to crash - maybe don't throw unless dealing with that in the component.